### DISCUSSION TITLE
keyglide: Helix/Neovim editor golf

### DISCUSSION THREAD

**Discussion by tomgroenwoldt** [helix-editor/helix#8359] [Show and tell]
Hey! I currently build a small [web platform](https://keygli.de) specifically for modal editor users which allows to host "challenges" [like this](https://github.com/helix-editor/helix/discussions/8130). A user can choose between helix or neovim to tackle the challenge.

https://github.com/helix-editor/helix/assets/70777530/963a35ae-5a6e-479e-9ba4-fe2b0b792a16

Feel free to give it a try! This is still in its very early stages. If you have any feedback, I'm happy to hear about that :)

**Comment by gabydd** [comment]
this looks great I'll probably add some interesting ones I have found from vimgolf, it might be nice to be able to add some kind of source so people know they aren't original, I couldn't find a license for using the actual challenges so I am not sure about how we are able to use it. I'll also try and come up with some original ones, hopefully that use a language so they can utilize some tree-sitter stuff as well.

**Reply by tomgroenwoldt** [reply]
Hi @gabydd! Thanks for the input, that's a great idea. From now on you have the possibility to specify an optional source (**but valid URL**) on challenge upload. On successful creation this is displayed like this:

![image](https://github.com/helix-editor/helix/assets/70777530/6968ef28-ded8-4ed6-9537-8e9dec210cb1)

**Comment by makemeunsee** [comment]
Hi @tomgroenwoldt,

Very promising project!
Do you have plans to release the source code, and/or make it open to contributions (beyond challenges)?
Also I wish there was a way to try it out without the email registration.
Looking forward to seeing it grow!

**Reply by David-Else** [reply]
Looks awesome, I just tried to sign up with a passphrase with an entropy of 103 bit but was greeted with the maddening:
```
Please enter a minimum of 12 characters with at least two uppercase, two lowercase, two digit and two symbol characters.
```
That advice is so out of date, please make it go away :) I would like to use a lower case pass phrase in case I actually need to type it, which does happen on occasion.

It would be nice to sign up with your GitHub account, most developers have one, it would save a lot of bother.

**Reply by tomgroenwoldt** [reply]
Hey @makemeunsee :) This is planned eventually, but this will take a bit of time. The plan would be a repository with frontend and backend completely separated from the infrastructure.

I currently look into something like a guest user. Sadly [authentik](https://github.com/goauthentik/authentik) doesn't support that yet if I am correct.

**Reply by tomgroenwoldt** [reply]
Thanks for the feedback @David-Else, this is true. I will look into that! :)

**Comment by David-Else** [comment]
Seems VERY promising! The main problems so far are:

- There are no instructions. How do I complete the challenge? `:w` does not do anything.
- Can we have a user esc key setting, for example `jk` is very common.
- How about a keystroke count as you play and the current high score?
- How do you see other peoples solutions? For me this would be the most valuable aspect.

Cheers, keep going, this is going to be amazing! I would be happy to get an email about updates, maybe letting new sign ups consent to getting emails regarding updates would be a good move?

**Reply by tomgroenwoldt** [reply]
Thanks, again, for the feedback! :)

- Right, this is a work in progress. `:w` + `Enter` should complete the challenge though. The file content has to be the exact same as the file shown on the right (including newline characters)
- I want to support user configurations in the future!
- Indeed, that would be very convenient.
- Currently, you can list all scores on a dedicated scores page or look into the top scores of a challenge respectively via the challenge page.

Great ideas! This is still some sort of prototype, the login system may change in the future.

**Comment by tomgroenwoldt** [comment]
Thanks for your feedback everyone! I will take this site offline as of now and continue working towards a more finished application :)

**Comment by tomgroenwoldt** [comment]
It's been a while and for those who are interested - please checkout https://keygli.de :)
